hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/types/NSSummary.java

```diff
@@ -93,7 +92,7 @@ public void setChildDir(Set<Long> childDir) {
   }
 
   public void setDirName(String dirName) {
-    this.dirName = dirName;
+    this.dirName = removeTrailingSlashIfNeeded(dirName);
```

Comment:
Double check the usage here.

IIRC if the target is a directory (not a key), removing the trailing slash might cause issue because of the key mismatch when getting value from RocksDB.

Something might have changes since then, if this is tested unaffected I'd be fine with this change.

Reply:
> IIRC if the target is a directory (not a key), removing the trailing slash might cause issue because of the key mismatch when getting value from RocksDB.

@smengcl This is the case for legacy, for FSO there is no difference. We wanted to handle legacy the same way as FSO and that's why we are removing the trailing slash. It's used by every test method, it's not making a difference. I can remove it and include it in the next PR.

Reply:
Yes, I believe it is correct and should be there for Legacy buckets.  A lot of legacy logic depends on the correct placement of that slash.  

Moving it to the legacy PR should be fine.

Reply:
Got it. I'm fine with this change. Thanks for the clarification!

Reply:
ah, I saw it is removed in the latest commit. In this case we can add it back in the actual legacy bucket support patch.